datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -881,6 +907,25 @@ pub fn replace_cols_by_name(
     })
 }
 
+/// check whether the expression uses the columns in `check_map`.
+pub fn contain(e: Expr, check_map: &HashMap<String, Expr>) -> Result<bool> {
+    let mut is_contain = false;
+    e.apply(&mut |expr| {
+        Ok(if let Expr::Column(c) = &expr {
+            match check_map.get(&c.flat_name()) {
+                Some(_) => {
+                    is_contain = true;
+                    VisitRecursion::Stop
+                }
+                None => VisitRecursion::Continue,
+            }
+        } else {
+            VisitRecursion::Continue
+        })
+    })?;
```

Comment:
Since the closure always returns `Ok` you could change this to `unwrap()` if you wanted and have `contain` return `bool` instead of `Result<bool>`